src/main/java/com/ldtteam/structurize/util/BlockUtils.java

```diff
@@ -386,4 +386,23 @@ else if (stackToPlace.getItem() == Items.WATER_BUCKET)
             world.setBlockState(here, Blocks.WATER.getDefaultState(), Constants.BlockFlags.BLOCK_UPDATE);
         }
     }
+
+    /**
+     * Removes the fluid from the given position.
+     * 
+     * @param world the world to remove the fluid from.
+     * @param pos   the position where to remove the fluid.
+     */
+    public static void removeFluid(World world, BlockPos pos)
+    {
+        final BlockState state = world.getBlockState(pos);
+        final Block block = state.getBlock();
+        if (block instanceof IBucketPickupHandler && ((IBucketPickupHandler)block).pickupFluid(world, pos, state) != Fluids.EMPTY)
+        {
```

Comment:
empty if?

Reply:
the block is getting emptied in the if, the if is just needed to detect whether it worked.

Reply:
as it doesn't work with non-source blocks, but they should get removed too, there has to still be the else, so i just have an empty if there.